### PR TITLE
Instanceof redirect and synthetic field shadowing tutorial

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Credits to @biom4st3r for the instanceof redirect example

### CODE REVIEW

File: shadow-anonymous.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+[<- Return](README.md)
+
+Allows accessing a 'local variable' used in an anonymous class, which is actually just a constant field.
+
+Shadowing the field is done the same way as a normal field, however finding the exact name of the field might be a bit tricky.
+
+### IntelliJ Idea
+- Open the class file
+
+![foo](https://i.imgur.com/hYWE1O3.png)
```
Comment (dblsaiko) [inline shadow-anonymous.md]: Could you put these inside the repo?

File: redirect.md [hunk 2]
```diff
@@ -36,4 +37,44 @@ Method modification:
...
+
+ - Object that will be the tested object
+ - The original Class being tested against
+ - Returns either a boolean or Class<?> your choice.
+
+**This conflicts with any other mixin targetting the same method call, field access, or object construction. Handle with care.**
+
+Example mixin:
+```java
+@Redirect(method = "target()Z", at = @At(value = "CONSTANT",args="classValue=net/example/Dummy",shift=Shift.AFTER,ordinal=0))
```
Comment (dblsaiko) [inline redirect.md]: Please format the code consistently (spaces before and after "=", spaces after ",", braces on same line, consistent indentation)

**Review by dblsaiko — CHANGES_REQUESTED** [review]
Interesting, didn't know you could redirect instanceof.

File: shadow-anonymous.md (hunk 1 above)
Comment (BoogieMonster1O1) [reply shadow-anonymous.md]: Done, ive put them in `/assets`

**Pull request merged**